### PR TITLE
Try running with latest PCT

### LINKED ISSUE #657: Use newer PCT
Currently using an incremental deployment of https://github.com/jenkinsci/plugin-compat-tester/pull/276. Need to get that released, and also check whether https://github.com/jenkinsci/plugin-compat-tester/pull/299 reveals failures (supposedly something in `workflow-scm-step at least).

### PR DESCRIPTION
**Pull request by jglick**
Fixes #657. Going to see what, if anything, https://github.com/jenkinsci/plugin-compat-tester/pull/276 was actually needed for. Also wondering if https://github.com/jenkinsci/plugin-compat-tester/pull/299 or other recent changes will break things.

### CODE REVIEW

**Comment by jglick** [comment]
Something weird in https://ci.jenkins.io/job/Tools/job/bom/job/PR-659/1/pipeline-console/?selected-node=263 and something else in `jquery3-api`; I suspect there is some sort of regression affecting plugins with no `src/test/java/` (just `InjectedTest`).

https://github.com/jenkinsci/bom/pull/659/checks?check_run_id=3733673007 seems to reproduce the need for https://github.com/jenkinsci/plugin-compat-tester/pull/276:

```
java.lang.NoSuchMethodError: io.jenkins.plugins.casc.ConfigurationAsCode.toYaml(Lio/jenkins/plugins/casc/model/CNode;)Lio/jenkins/plugins/casc/snakeyaml/nodes/Node;
	at io.jenkins.plugins.casc.misc.Util.toYamlString(Util.java:152)
	at hudson.plugins.view.dashboard.ConfigurationAsCodeBackCompatTest.testOldImagePortletUrl(ConfigurationAsCodeBackCompatTest.java:63)
```

**Comment by jglick** [comment]
TBD if only `dashboard` is affected. If the problem is just an old JCasC dep, then fine. But AFAICT it is using something newish.

**Comment by jglick** [comment]
Hmm, `failFast` not working any more? (In this case, good.)

**Comment by jglick** [comment]
850f995cf2d414f4c77ffc8a7ad54cbfbca3c874 might have broken `failFast`, TBD.

**Comment by jglick** [comment]
Well, maybe it does still work, just more slowly.

**Comment by timja** [comment]
> if dashboard is the only plugin with this issue why not just fix that plugin instead of creating an hack?

from what i remember it's not a hack as such, the configuration as code plugin and test harness are not correctly update by PCT, only the plugin is updated and not the test harness.

It creates a nightmare when they aren't updated together.

not such an issue now while there's not many changes but still has been a huge pain before

**Comment by jglick** [comment]
Perplexed by why https://ci.jenkins.io/job/Tools/job/bom/job/PR-659/2/testReport/hudson.plugins.view.dashboard/ConfigurationAsCodeTest/ would be passing against the latest line yet failing against the oldest line. The Jenkins version should not make any difference, and we are using the same `configuration-as-code` version in all lines. (And same for `snakeyaml-api`.)

**Comment by jglick** [comment]
There is also a PCT regression affecting `workflow-scm-step` ([log](https://ci.jenkins.io/job/Tools/job/bom/job/PR-659/2/execution/node/3165/log/)):

```
 [ERROR]     'dependencies.dependency.version' for org.jenkins-ci.plugins.workflow:workflow-step-api:jar is missing. @ line 66, column 17
```

**Comment by jglick** [comment]
In sum, there are at least PCT bugs visible in this project:
* `configuration-as-code` test harness issue, as already suspected
* library wrapper plugins broken
* something involving `workflow-scm-step`

**Comment by imonteroperez** [comment]
> In sum, there are at least three PCT bugs visible in this project:
> 
> * `configuration-as-code` test harness issue, as already suspected
> * library wrapper plugins broken
> * something involving `workflow-scm-step`

Regarding `workflow-scm-step` I can confirm it was failing with the same issue before https://github.com/jenkinsci/plugin-compat-tester/pull/299 gets merged

File: prep.sh [hunk 1]
```diff
@@ -41,7 +41,7 @@ do
 done
 
 # TODO find a way to encode this in some POM so that it can be managed by Dependabot
-version=0.5.1-rc1066.9cfd30140d30 # TODO https://github.com/jenkinsci/plugin-compat-tester/pull/276
+version=0.5.1-rc1106.47640a4e521e
```
Comment (imonteroperez) [inline prep.sh:44]:
```suggestion
version=0.5.1
```
https://github.com/jenkinsci/plugin-compat-tester/releases/tag/plugins-compat-tester-aggregator-0.5.1

Comment (jglick) [reply prep.sh]: We still need https://github.com/jenkinsci/plugin-compat-tester/pull/276.

**Comment by jglick** [comment]
https://github.com/jenkinsci/plugin-compat-tester/pull/320 ought to speed things up a bit.

**Comment by jglick** [comment]
#1336

**Pull request closed without merging**